pkg/nodeagent/controller/operatingsystemconfig/changes.go

```diff
@@ -80,6 +80,12 @@ type files struct {
 func computeOperatingSystemConfigChanges(fs afero.Afero, newOSC *extensionsv1alpha1.OperatingSystemConfig) (*operatingSystemConfigChanges, error) {
 	changes := &operatingSystemConfigChanges{}
 
+	// osc.files and osc.unit.files should be changed the same way by OSC controller.
+	// The reason for assigning files to units is the detection of changes which require the restart of a unit.
+	newUnitFiles := collectUnitFiles(mergeUnits(newOSC.Spec.Units, newOSC.Status.ExtensionUnits))
+	newOSCFiles := append(newOSC.Spec.Files, newOSC.Status.ExtensionFiles...)
+	newOSCFiles = append(newOSCFiles, newUnitFiles...)
```

Comment:
Maybe make a function for this since the exact same lines are needed again below for the old OSC